docs/styleguide/generic.rst

```diff
@@ -162,3 +162,43 @@ Comments
       do_something()
 
 * Avoid end of line comments.
+
+Git
+===
+
+Feature branch development
+--------------------------
+
+The code of our project is shared by all members of the team.
+
+We often have a feature branch that is driven by a single developer, but during
+the branch lifetime or the review process that branch may get contributions from
+other developers. For example typo-fixes or small GUI fixes.
+
+To reduce conflicts and confusion between colleagues, avoid using git commands
+that rewrite the history (rebase or commit with amend).
+
+See  :ref:`here on how to keep a feature branch up to date
+<keep-feature-branch-up-to-date>`.
+
+Don't revert, just push the fix
+-------------------------------
+
+If you find an issue after merging into the main branch, avoid the trouble of
+reverting the commit before pushing a fix.
+
+If it is clear that the commit is broken (like showing a failing test), and the
+team will not release it, then feel free to just push the fix on top.
+
+Delete obsolete branches
+------------------------
+
+Branches you no longer need can be a nuisance, both on your machine and on
+GitHub, by polluting auto-complete and screen space. These can be:
+
+* Branches that were already merged,
+* Branches for which the code is already in the main branch or another branch,
+  even if the branch itself was not merged, or
+* Obsolete branches on which we will never work in the future.
+
+`Delete them <https://stackoverflow.com/a/2003515>`_.
```

Comment:
Must include the required commands. Then you can also link to other sites for more info, if so inclined.

Reply:
All right. Though Adi [said](https://github.com/chevah/geck/pull/79#discussion_r566375399) we shouldn't have full Git tutorials in here, the commands for deletion are not that obvious.

Thank you for the feedback!

Reply:
Yes... Not full tutorial, but it's OK to have quick command as reference especially if this commented by a reviewer :)


Thanks for the update.

And yes... amend only for pushed commits.

You can do whatever you want on your local branch that is not synced with GitHub :)